### PR TITLE
Datasource/Loki: Loki now goes to Logs mode when importing prom queries

### LINKED ISSUE #20831: Loki: Query import from Prometheus going to metrics mode
0. Explore with Prometheus and Loki
1. Select Prometheus and run a query with labels that exist both in Loki and Prometheus
2. Switch to Loki

Result: Loki imported the query, but is started in Metrics mode so the query returns an error. 

Expected: Prometheus-to-loki switch should start in Logs mode since I'm expecting to find logs for the metrics in the majority of cases.

### PR DESCRIPTION
**Pull request by kaydelaney**
**What this PR does / why we need it**:
Loki now goes to Logs mode when importing prom queries.

**Which issue(s) this PR fixes**:
Closes #20831

### CODE REVIEW

**Review by davkal — COMMENTED** [review]
Switch from prom to loki logs works now, but it broke initializion from url:
load loki with metrics, then refresh browser -> logs mode.

Is there a way to write unit-y tests for both behaviors? Otherwise this could be something for e2e tests.

**Comment by kaydelaney** [comment]
ack, yes, good idea

File: public/app/features/explore/state/actions.test.ts [hunk 1]
```diff
@@ -163,6 +173,62 @@ describe('refreshExplore', () => {
...
+      testDatasource: () => Promise.resolve({ status: 'success' }),
+      name,
+      init: jest.fn(),
+      meta: { id: 'some id', name },
+    };
+
+    const initialState = {
+      explore: {
+        [exploreId]: {
+          requestedDatasourceName: name,
```
Comment (davkal) [inline public/app/features/explore/state/actions.test.ts]: Shouldn't the requested ds be Loki?

**Review by davkal — APPROVED** [review]
Behavior and code is ace!

File: public/app/features/explore/state/actions.test.ts (hunk 1 above)
Comment (kaydelaney) [reply public/app/features/explore/state/actions.test.ts]: good catch, hold on
```suggestion
          requestedDatasourceName: 'Loki',
```

**Pull request merged**